magento/magento2
# Removing unused legacy price code

### Description

Removing some legacy code that doesn't ever look used in m2. There don't appear to be ant elements with `#price_default` in the code. This is one of the many uses of `Event.observe` which hopefully we can move towards removing.

The code doesn't appear to have tests or any other remaining structure relating to it.

### Manual testing scenarios
1. Check anything that would contain prices, however as mentioned I can't see anything that would trigger this code.

### Contribution checklist
